controller/src/beerocks/bml/internal/bml_internal.cpp

```diff
@@ -1874,8 +1910,38 @@ int bml_internal::client_get_client_list(char *client_list, unsigned int *client
     // Clear the promise holder
     m_prmClientListGet = nullptr;
 
-    LOG_IF(iRet != BML_RET_OK, ERROR)
-        << "Get client list request returned with error code:" << iRet;
+    if (iRet != BML_RET_OK) {
+        LOG(ERROR) << "Get client list request returned with error code:" << iRet;
+        return (iRet);
+    }
+
+    bool result = prmClientListGet.get_value();
+    LOG(DEBUG) << "Promise resolved, received " << client_mac_list.size() << " clients";
+    if (!result) {
+        LOG(ERROR) << "Get client list request failed";
+        return -BML_RET_OP_FAILED;
+    }
+
+    // Translate sMacAddr list to string
+    std::string client_list_str;
+    const std::string delimiter(",");
+    for (const auto &client : client_mac_list) {
+        auto mac_str = tlvf::mac_to_string(client);
+        if (client_list_str.size() + mac_str.size() + delimiter.size() >= client_list_max_size) {
+            break;
+        }
+        client_list_str.append(mac_str + delimiter);
+    }
+
+    // Remove the last delimiter
+    if (!client_list_str.empty()) {
+        client_list_str.pop_back();
+    }
```

Comment:
If you will take my suggestion, you could remove it.

Reply:
replaced the `pop_back()` with `string.back() = '\0'`